src/Shell/Task/ModelTask.php

```diff
@@ -691,14 +691,14 @@ public function getValidation($model, $associations = [])
      * @param \Cake\Database\Schema\TableSchema $schema The table schema for the current field.
      * @param string $fieldName Name of field to be validated.
      * @param array $metaData metadata for field
-     * @param string $primaryKey The primary key field
+     * @param array $primaryKey The primary key field
      * @return array Array of validation for the field.
      */
-    public function fieldValidation($schema, $fieldName, array $metaData, $primaryKey)
+    public function fieldValidation($schema, $fieldName, array $metaData, array $primaryKey)
```

Comment:
I don't think this is a backwards compatible change. Type hint change for `$primaryKey` var will cause error if method is overridden in an extending class with old signature.

Reply:
In this case this is probably fair enough as it was always internally expected to be an array (or crash).
See code. This just puts the error more clearly to the beginning of the method.

Reply:
Even if one is correctly passing an array he would still get an error due to method signature mismatch in child class, which shouldn't happen.